modules/33acrossBidAdapter.js

```diff
@@ -553,14 +617,19 @@ function _isIframe() {
 }
 
 // **************************** INTERPRET RESPONSE ******************************** //
-// NOTE: At this point, the response from 33exchange will only ever contain one bid
-// i.e. the highest bid
 function interpretResponse(serverResponse, bidRequest) {
   const bidResponses = [];
-
-  // If there are bids, look at the first bid of the first seatbid (see NOTE above for assumption about ttx)
-  if (serverResponse.body.seatbid.length > 0 && serverResponse.body.seatbid[0].bid.length > 0) {
-    bidResponses.push(_createBidResponse(serverResponse.body));
+  const { seatbid, cur } = serverResponse.body;
+
+  if (utils.isArray(seatbid)) {
+    seatbid.forEach(function(seat) {
+      if (utils.isArray(seat.bid)) {
```

Comment:
I am also seeing an opportunity here to split the code and eliminate the nested branching, 
e.g. a new function that creates the bid responses for a seatbid
```
let bidResponses = [];
if (utils.isArray(seatbid)) {
  seatbid.forEach(function(seat) {
    bidResponses = bidResponses.concat(createBidResponsesForSeat(seat))
    // or bidResponses.push(...createBidResponsesForSeat(seat));
  }
}  
```

Reply:
Hmm, what is the benefit of de-nesting? I ask since the logic here is simple and will not ever be extended to include other conditions (unless ORTB structure itself changes). Also to give you some context, in the programming style we follow, the helper functions never have conditionals in them. The caller is responsible to decide when to invoke them. Of course, when the decision making logic is complex it is beneficial to break it off into a separate helper. This article may serve as reference https://www.digitalocean.com/community/posts/5-tips-to-write-better-conditionals-in-javascript

Let me know your thoughts. If you are not convinced, I will eliminate double nesting.

Reply:
Got it, I like the idea of not having conditionals inside helpers, an alternative to remove the nested conditional could be an early return, something suggested in the article you sent.
e.g
```
if (!utils.isArray(seatbid)) {
    return bidResponses;
}
```

re: why to do the de-nesting, I like to make the code more scalable, probably the structure won't be changed but using a helper and/or do an early return makes the code easier to read from my perspective.

Reply:
Yea, let me do the above.

Reply:
So what I have done is a bit different than what we discussed above. Given that we use elements of functional programming I have utilized Array functions to filter and map into desired structure. Not broken the callbacks into separate functions as yet since the logic is way to simple, but it allows for us to expand into separate helpers if need be without adding complicated nested logic. Let me know your thoughts on the updated version